pkg/sqs/sqs.go

```diff
@@ -0,0 +1,127 @@
+package sqs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+	sourcesdk "github.com/numaproj/numaflow-go/pkg/sourcer"
+	"log"
+	"strconv"
+	"sync"
+	"time"
+)
+
+const (
+	ApproximateNumberOfMessages           = "ApproximateNumberOfMessages"
+	ApproximateNumberOfMessagesNotVisible = "ApproximateNumberOfMessagesNotVisible"
+	MaxNumberOfMessages                   = 10 // MaxNumberOfMessages is messages to return From SQS Valid values: 1 to 10. Default: 1
+	WaitTimeSeconds                       = 20 // WaitTimeSeconds specifies the time (in seconds) the call waits for a message to arrive in the queue before returning. If a message arrives, the call returns early; if not and the time expires, it returns an empty message list.
+
+)
+
+// AWSSqsSource represents an AWS SQS source with necessary attributes.
+type AWSSqsSource struct {
+	lock             *sync.Mutex
+	queueURL         *string
+	toAckSet         map[string]struct{}
+	sqsServiceClient sqsiface.SQSAPI
+}
+
+// NewAWSSqsSource creates a new AWSSqsSource using the given AWS configuration.
+func NewAWSSqsSource(sqsServiceClient sqsiface.SQSAPI, queueName string) (*AWSSqsSource, error) {
+	url, err := sqsServiceClient.GetQueueUrl(&sqs.GetQueueUrlInput{QueueName: aws.String(queueName)})
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Error in Getting Queue URL %s", err))
+	}
+	return &AWSSqsSource{
+		lock:             new(sync.Mutex),
+		queueURL:         url.QueueUrl,
+		sqsServiceClient: sqsServiceClient,
+	}, nil
+}
+
+func (s *AWSSqsSource) GetApproximateMessageCount(queryType string) int64 {
```

Comment:
Please comment on what `GetApproximateMessageCount` does. For someone who is not familiar with the concept. It would be even better if we explain the rationale behind why we choose to use it as Pending.